CMakeLists.txt.ros2

```diff
@@ -0,0 +1,248 @@
+cmake_minimum_required(VERSION 3.5)
+project(voxelized_geometry_tools)
+
+find_package(ament_cmake_ros REQUIRED)
+find_package(geometry_msgs REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(rosidl_default_generators REQUIRED)
+find_package(sensor_msgs REQUIRED)
+find_package(std_msgs REQUIRED)
+find_package(visualization_msgs REQUIRED)
+
+find_package(common_robotics_utilities REQUIRED)
+
+find_package(Eigen3 REQUIRED)
+set(Eigen3_INCLUDE_DIRS ${EIGEN3_INCLUDE_DIR})
+find_package(OpenMP)
+find_package(CUDA)
+find_package(OpenCL)
+
+#######################################
+## Declare ROS messages and services ##
+#######################################
+
+## Generate messages in the 'msg' folder
+
+rosidl_generate_interfaces(
+  ${PROJECT_NAME}_interfaces
+  msg/CollisionMapMessage.msg
+  msg/DynamicSpatialHashedCollisionMapMessage.msg
+  msg/SignedDistanceFieldMessage.msg
+  msg/TaggedObjectCollisionMapMessage.msg
+  DEPENDENCIES std_msgs
+)
+
+###########
+## Build ##
+###########
+
+## Specify additional locations of header files
+## Your package locations should be listed before other locations
+include_directories(include SYSTEM ${Eigen3_INCLUDE_DIRS})
+
+## Build options
+add_compile_options(-std=c++14)
+add_compile_options(-Wall)
+add_compile_options(-Wextra)
+add_compile_options(-Werror)
+add_compile_options(-Wconversion)
+# add_compile_options(-Wshadow)
```

Comment:
Are these options commented out for a reason?

Reply:
I forgot to comment about these. `rclcpp` has a few... hmm, rough edges that weren't fully fleshed out for Dashing and Eloquent. These have been fixed in Foxy IIRC. So as long as we support those distros (and we need Eloquent), we can't afford these warnings. It's not ideal, I know.

Reply:
I assume these warnings are coming in via the generated message code? If there's a way to turn these warnings off for *only* the included ROS2 code that would be ideal.

Reply:
> I assume these warnings are coming in via the generated message code?

Not only, no.

> If there's a way to turn these warnings off for only the included ROS2 code that would be ideal.

There isn't. But https://github.com/ament/ament_cmake/pull/297 just got in, which can help us workaround this issue. I'll talk to the corresponding ROS bosses, see if we can get it backported.

Reply:
Alright, backporting conversations started. 

I can't promise it'll be fast though. CC @EricCousineau-TRI.

Reply:
Supporting a `SYSTEM` include keyword would be ideal. These warnings were added because drake and anzu use them, and having to disable them there is not going to fly.

Reply:
Any update on the backporting work?

Reply:
Sorry for the long round-trip. https://github.com/ament/ament_cmake/pull/297 was backported in https://github.com/ament/ament_cmake/pull/307, but it was just last week that an Eloquent sync happened.

As of https://github.com/calderpg/voxelized_geometry_tools/pull/6/commits/bf0d58cf977c408f77d808f087e349619d70c8b1, all warnings are back up.